### PR TITLE
Feat push notification

### PR DESCRIPTION
**Pull request by ruveyda-kisla**
### Description

- add pwa support
- add firebase and notification example

### Checklist

- [x] Tested changes locally.
- [x] Ensured code follows project standards.
- [x] Updated documentation.
- [x] Added necessary tests.

### CODE REVIEW

File: public/firebase-messaging-sw.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+function sendMessageToAllClients(command, message) {
+  clients.matchAll({ type: 'window' }).then(function (windowClients) {
```
Comment (ceydae) [inline public/firebase-messaging-sw.js:2]: console.log clients and see what that is

File: public/firebase-messaging-sw.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        });
+  }
+});
+
+self.addEventListener('push', function (event) {
+  if (event.data) {
+    const data = event.data.json();
+    const title = data.notification?.title;
+    const body = data.notification?.body;
+    const iconUrl = './images/wagentLogo.svg';
```
Comment (ceydae) [inline public/firebase-messaging-sw.js:58]: change iconUrl to an absolute url

File: public/sw.js
Comment (ceydae) [inline public/sw.js:1]: is this file needed?

File: public/sw.js.map
Comment (ceydae) [inline public/sw.js.map:1]: is this needed?

File: src/app/layout.tsx
Comment (ceydae) [inline src/app/layout.tsx:1]: is this line needed?

File: src/app/page.tsx
Comment (ceydae) [inline src/app/page.tsx:1]: remove test component

File: src/containers/Test/index.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+'use client';
+
+import React from 'react';
+import { requestNotificationPermission } from '@/services/firebase';
+import CButton from '@/components/CButton';
+
+const Test = () => {
+  const handleClick = () => {
+    requestNotificationPermission();
```
Comment (ceydae) [inline src/containers/Test/index.tsx:9]: Call this in a file that if the page loads, it calls this function. 

I don't know what is the appropriate file is in Next 14 though

File: src/services/firebase.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+          token: token,
+          title: 'Wagent',
+          body: 'An Item has just been Sold!',
+        }),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+    } else {
+      console.log('No registration token available. Request permission to generate one.');
```
Comment (ceydae) [inline src/services/firebase.ts:49]: change the `getToken` to something like `sendClientToken` and send the token to the back-end to store for future use.

**Review by ceydae — CHANGES_REQUESTED** [review]

File: src/app/layout.tsx
Comment (ruveyda-kisla) [reply src/app/layout.tsx:1]: if you mean manifest meta tag 
Yes

**Review by ceydae — APPROVED** [review]

**Pull request merged**